src/PrivateRoute.jsx

```diff
@@ -0,0 +1,12 @@
+import { useSelector } from 'react-redux';
+import { getUser } from './store/selectors';
+import { Navigate } from 'react-router-dom';
+
+const PrivateRoute = ({ children }) => {
+	const user = useSelector(getUser);
+
+	const isAdmin = user?.role?.toLowerCase() === 'admin';
```

Comment:
Will be better to create one selector for that(selectIsUserAdmin) and reuse it in the application and keep that logic in one place